cli/lockfile.rs

```diff
@@ -30,19 +45,41 @@ pub struct Lockfile {
 
 impl Lockfile {
   pub fn new(filename: PathBuf, write: bool) -> Result<Lockfile, AnyError> {
-    let map = if write {
-      BTreeMap::new()
+    // Writing a lock file always uses the new format.
+    let content = if write {
+      LockfileContent::V2(LockfileV2Content {
+        version: "2".to_string(),
+        remote: BTreeMap::new(),
+      })
     } else {
       let s = std::fs::read_to_string(&filename).with_context(|| {
         format!("Unable to read lockfile: {}", filename.display())
       })?;
-      serde_json::from_str(&s)
-        .context("Unable to parse contents of the lockfile")?
+      let value: serde_json::Value = jsonc_parser::parse_to_serde_value(
```

Comment:
We should probably keeps this using serde_json. Allowing jsonc for the lockfile doesn't seem necessary?

Reply:
Oh, I see it's for the leading comments. The one issue with having comments is then it makes it harder for some other tools to read the lockfile. I think what's said in the comments is somewhat self explanatory with a lockfile?

Reply:
Yeah, maybe, but other tools usually add such comments (though they often use a file format that supports comments OOTB